cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -873,7 +875,7 @@ Status TransferDate64(RecordReader* reader, MemoryPool* pool,
 Status TransferDictionary(RecordReader* reader,
                           const std::shared_ptr<DataType>& logical_value_type,
                           std::shared_ptr<ChunkedArray>* out) {
-  auto dict_reader = dynamic_cast<internal::DictionaryRecordReader*>(reader);
+  auto dict_reader = dynamic_cast<DictionaryRecordReader*>(reader);
```

Comment:
Might as well replace it with checked_cast.

Reply:
Indeed.

Reply:
Actually, no, the base class is virtual, so `dynamic_cast` is required.